bundles/specmate-integration-test/src/com/specmate/test/integration/CauseEffectPatternTest.java

```diff
@@ -0,0 +1,99 @@
+package com.specmate.test.integration;
+
+import static org.junit.Assert.assertEquals;
+
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.uima.jcas.JCas;
+import org.eclipse.emf.common.util.URI;
+import org.junit.Assert;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+
+import com.specmate.cause_effect_patterns.parse.DependencyParsetree;
+import com.specmate.cause_effect_patterns.parse.matcher.MatchResult;
+import com.specmate.cause_effect_patterns.parse.matcher.MatchRule;
+import com.specmate.cause_effect_patterns.parse.matcher.MatchUtil;
+import com.specmate.cause_effect_patterns.parse.matcher.Matcher;
+import com.specmate.cause_effect_patterns.parse.matcher.MatcherException;
+import com.specmate.cause_effect_patterns.resolve.XTextUtil;
+import com.specmate.common.exception.SpecmateException;
+import com.specmate.nlp.api.ELanguage;
+import com.specmate.nlp.api.INLPService;
+
+public class CauseEffectPatternTest {
+	
+	@Test
+	public void testLoadRules() throws SpecmateException, URISyntaxException {
+		List<MatchRule> res = loadRules("/resources/rule_EN.spec");
+		assertEquals(35, res.size());
+		
+		List<MatchRule> res2 = loadRules("/resources/test_ruleset.spec");
+		assertEquals(2, res2.size());
+	}
+	
+	
+	@Test
+	public void testDependencyParse() throws SpecmateException, MatcherException, URISyntaxException {
+		INLPService nlpService = NLPServiceTest.getNLPService();
+		JCas result = nlpService.processText("If the tool encounters an error then it beeps.", ELanguage.EN);
+		
+		DependencyParsetree data = DependencyParsetree.generateFromJCas(result);
+		Assert.assertEquals(data.getHeads().size(), 1);
+		
+		List<Matcher> rules = convertMatchRule(loadRules("/resources/test_ruleset.spec"));
+		
+		// Run the rules
+		List<MatchResult> results = MatchUtil.evaluateRuleset(rules, data);
+		Assert.assertEquals(data.getHeads().size(), results.size());
+		
+		MatchResult res = results.get(0);
+		Assert.assertTrue(res.isSuccessfulMatch());
+		
+		// Get the result:
+		//Cause
```

Comment:
Can you add an assert that checks if the correct text has been extracted? The same holds for the effect